shell/renderer/api/electron_api_context_bridge.cc

```diff
@@ -464,17 +508,16 @@ void ProxyFunctionWrapper(const v8::FunctionCallbackInfo<v8::Value>& info) {
 
   {
     v8::Context::Scope func_owning_context_scope(func_owning_context);
-    context_bridge::ObjectCache object_cache;
```

Comment:
I know this was done as the result of one of my own comments but annoyingly this would be a breaking change. It would break object identity expectations within a single stack. i.e. imagine the function `(a, b) => a === b` would break for the input `const foo = {}; fn(foo, foo)` as it would clone twice and not reuse the cache.

In this casce we can maintain original behavior by calling `PassValueToOtherContext` on the original array I guess? Or calling the Inner method such that we can provide our own object cache to retain previous caching semantics

Reply:
Good catch! I may allow `PassValueToOtherContext` to optionally provide the object cache or call the inner method directly. I think this issue would occur with the args of `executeInMainWorld` as well.

Reply:
Fixed for proxied function arguments and `executeInMainWorld` arguments in [c8aa249](https://github.com/electron/electron/pull/44411/commits/c8aa24969e746821c107a7f246e7abb2053132be). Also added tests for both.